### ISSUE TITLE
Accessibility needs fixing (currently completely broken)

### ISSUE THREAD

**Issue by chris-morgan** [jekuer/add-to-calendar-button#28]
This is currently completely inaccessible to screen readers, keyboard users and such, because you’ve built it on unfocusable divs and mouse hover and touch events. For accessibility purposes: (a) never depend only on those events; (b) never use an unfocusable element—at the least, give the div a tabindex=0 attribute; (c) don’t use divs where you mean buttons or links.

I believe this library could do with being split into two parts, one for functionality and one for DOM; this would allow alternative implementations of the DOM half to fit in with existing UI frameworks. This may be beyond the effort you wish to go to; up to you. But the accessibility parts at least of the current implementation really need fixing.

**Comment by jekuer** [comment]
This is discussable and therefore I would consider "completely broken" a little harsh ;).
Switching to <button> should be a first good step into the right direction.

Enabling the functionality to navigate through it with the keyboard only brings in definitely more complexity since it was never thought as a native dropdown. I see three ways here:

1. observing keyboard input and translate it to some smart behavior (example: https://codepen.io/emmawedekind/pen/eXNppG).
2. rebuilding the tool as dropdown - maybe only optional (controlled by an additional JSON flag).
3. Going with something like https://codepen.io/lisseteb/pen/LLovRO

Since, as said, it is no real dropdown (you do not select something), I tend to go with option 3.
